### PR TITLE
Add API specification for chs-delta-api

### PR DESCRIPTION
**Pull request by amoriartyCH**
Open API 3.0 implementation using yaml
structure implementation of all known endpoints
and schema objects

### CODE REVIEW

File: apispec/schema.yml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+          maximum: 50
+        corporate_ind:
+          type: boolean
+        surname:
+          type: string
+          maximum: 160
+        forename:
+          type: string
+          maximum: 50
+        middle_namename:
```
Comment (mreis-ch) [inline apispec/schema.yml:67]: Typo? namename instead of name

File: apispec/schema.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        - officers
+        - CreatedAt
+        - delta_at
+
+    Officer:
+      type: object
+      properties:
+        company_number:
+          type: string
+        changed_at:
```
Comment (ch-tstrickland) [inline apispec/schema.yml:48]: Could add format restriction here:
```
        changed_at:
          type: string
          format: date-time
```

File: apispec/schema.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          type: string
+        kind:
+          type: string
+        internal_id:
+          type: string
+        appointment_date:
```
Comment (ch-tstrickland) [inline apispec/schema.yml:54]: Could add:
`format: date`

File: apispec/api-spec.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+openapi: 3.0.3
+info:
+  title: CHS Delta API
+  description: API specification for the chs-delta-api service
+  version: "1.0"
+
+paths:
+  /delta/officer-delta:
```
Comment (divya051988) [inline apispec/api-spec.yml:8]: The path should be /delta/officers please .

File: apispec/officer-delta-spec.yml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      application/json:
+        schema:
+          $ref: '#/components/schemas/Officer_delta'
+  responses:
+    '200':
+      description: Successfully produced message onto Kafka topic.
+    '400':
+      description: Bad request body - validation errors.
+    '401':
+      description: Unauthorised - missing api key in header.
```
Comment (divya051988) [inline apispec/officer-delta-spec.yml:15]: Should we include a 500 as well for Internal server error ?

File: apispec/officer-delta-spec.yml [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+          type: string
+        previous_forename:
+          type: string
+        previous_timestamp:
+          type: string
+
+    Identification:
+      type: object
+      properties:
+        EEA:
```
Comment (divya051988) [inline apispec/officer-delta-spec.yml:122]: The Identification has ""EEA"/"non_eea"/"UK_limited_company"/"other_corporate_body_or_firm" based on mock delta schemas- https://companieshouse.atlassian.net/wiki/spaces/DataS/pages/3229745153/MOCK+Delta+Schema+Officer+URA+Validation

File: apispec/officer-delta-spec.yml [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+          $ref: '#/components/schemas/EEA'
+
+    EEA:
+      type: object
+      properties:
+        place_registered:
+          type: string
+        registration_number:
+          type: string
+      required:
```
Comment (divya051988) [inline apispec/officer-delta-spec.yml:132]: Would we need legal_authority and legal_form ?

**Comment by divya051988** [comment]
Could we also update routes.yml to specify /delta/officers instead of allowing /delta/* ?

**Review by divya051988 — APPROVED** [review]

**Pull request merged**